src/lib/block_producer/block_producer.ml

```diff
@@ -650,6 +654,12 @@ let run ~context:(module Context : CONTEXT) ~vrf_evaluator ~prover ~verifier
         | None ->
             log_bootstrap_mode () ; Interruptible.return ()
         | Some frontier -> (
+            let global_slot =
+              Consensus.Data.Block_data.global_slot_since_genesis block_data
+            in
+            Internal_tracing.with_slot global_slot
```

Comment:
Should be negligible. It just attaches the block id to the execution context by calling `Async_kernel.Async_kernel_scheduler.with_local`.

Reply:
Btw, the reason this is needed is that because blocks can be processed concurrently, context switches need to be detected and recorded to be able to properly associate the checkpoint to the correct block.